source/code/scxsystemlib/disk/statisticallogicaldiskenumeration.cpp

```diff
@@ -234,6 +234,12 @@ namespace SCXSystemLib
             SCXCoreLib::SCXHandle<StatisticalLogicalDiskInstance> disk = *iter;
             disk->Update();
 
+//Fix for Bug10019956
+#if defined(sun)
+        if(disk->m_fsType == L"zfs" && (disk->m_device).find(L"/") != -1 && (disk->m_device).find(L"/") != 0)
```

Comment:
The logic check here is:

1. If the disk filesystem is zfs, 
2. and a "/" is found in the m_device string,
3. and the first character in the m_device string is not a "/",
4. Then, skip this disk

I don't have enough context to know if this is the right thing to do or not, but I agree with Jeff's comments, it would be nice to know what this logic check is intended to filter out, with an example.

Moreover, I think checking the result of find for -1 is incorrect, although it will probably work, but the proper check is if "find" returns "wstring::npos".